script/model/forecast.py

```diff
@@ -0,0 +1,520 @@
+
+        split_idx = len(data) - self.test_set_size
+        # First part as the training set
+        train_raw_data = data[:split_idx]
+
+        # Last part as the testing set
+        test_raw_data = data[split_idx:]
+
+        return train_raw_data, test_raw_data
+
+    def create_seqs(self) -> List[Tuple[torch.Tensor, torch.Tensor]]:
+        """
+        Create training sequences.  Each training sample consists of:
+         - A list of data points as input
+         - A label (target) value in some horizon
+        :return: training sequences
+        """
+        seq_len = self.seq_len
+        horizon = self.horizon_len
+        train_norm_data = self._normalize_data(self.train_raw_data)
+
+        input_data = torch.FloatTensor(train_norm_data).view(-1)
+        seqs = []
+        for i in range(len(input_data) - seq_len - horizon):
+            seq = input_data[i:i + seq_len]
+
+            # Look beyond the horizon to get the label
+            label_i = i + seq_len + horizon
+            label = input_data[label_i: label_i + 1]
+
+            seqs.append((seq, label))
+
+        return seqs
+
+    def get_test_input(self) -> List[float]:
+        """
+        Get the test input (normalized)
+        :return: test time-series data
+        """
+        test_input_norm = self._normalize_data(self.test_raw_data)
+        test_input_list = test_input_norm.tolist()
+        return test_input_list
+
+    def stat_util(self) -> None:
+        trace_start_ms = self.data[0][1]
+        trace_end_ms = self.data[-1][1]
+        trace_nsample = len(self.data)
+
+        print(
+            f"{trace_nsample} samples from {trace_start_ms} to {trace_end_ms}\n"
+            f"{(trace_end_ms - trace_start_ms) / 1000000}seconds")
+
+
+class LSTM(nn.Module):
+    """
+    A simple LSTM that serves as a placeholder for the training model
+    """
+
+    def __init__(self, input_size=1, hidden_layer_size=100, output_size=1):
+        """
+        :param input_size: One data point that is fed into the LSTM each time
+        :param hidden_layer_size:
+        :param output_size: One output data point
+        """
+        super().__init__()
+        self.hidden_layer_size = hidden_layer_size
+
+        self.lstm = nn.LSTM(input_size, hidden_layer_size)
+
+        self.linear = nn.Linear(hidden_layer_size, output_size)
+
+        self.hidden_cell = (torch.zeros(1, 1, self.hidden_layer_size),
+                            torch.zeros(1, 1, self.hidden_layer_size))
+
+    def forward(self, input_seq: torch.FloatTensor) -> float:
+        """
+        Forward propogation
+        :param input_seq:  1D FloatTensor
+        :return: A single value prediction
+        """
+        lstm_out, self.hidden_cell = self.lstm(
+            input_seq.view(len(input_seq), 1, -1), self.hidden_cell)
+        predictions = self.linear(lstm_out.view(len(input_seq), -1))
+        return predictions[-1]
+
+    def do_train(self, data_loader: DataLoader,
```

Comment:
Don't pass the `DataLoader` class here. Instead, generate the data outside and directly pass the data in here.